src/sciline/visualize.py

```diff
@@ -82,28 +113,72 @@ def _to_subgraphs(graph: FormattedGraph) -> dict[str, FormattedGraph]:
+        else:
+            name = f'<{ret_name}>'
+        if mode == 'data' and formatted_p.kind == 'function':
+            # Show provider name in data mode
+            via = f'<font point-size="11">via:<i>{formatted_p.name}</i></font>'
+            if common_provider:
+                origin = ret_name.split('[')[0]
+                subgraph.attr(label=f'<{origin}<br/>{via}>')
+            else:
+                name = f'{name[:-1]}<br/>{via}>'
+        shape = 'box3d' if formatted_p.ret.collapsed else 'rectangle'
         if formatted_p.kind == 'unsatisfied':
             subgraph.node(
-                formatted_p.ret.name,
-                formatted_p.ret.name,
-                shape='box3d' if formatted_p.ret.collapsed else 'rectangle',
+                ret_name,
+                name,
+                shape=shape,
                 color='red',
-                fontcolor='red',  # Set text color to red
+                fontcolor='red',
                 style='dashed',
             )
-        else:
-            subgraph.node(
-                formatted_p.ret.name,
-                formatted_p.ret.name,
-                shape='box3d' if formatted_p.ret.collapsed else 'rectangle',
-            )
+        elif mode != 'task' or formatted_p.kind == 'parameter':
+            subgraph.node(ret_name, name, shape=shape)
         if formatted_p.kind == 'function':
-            dot.node(p, formatted_p.name, shape='ellipse')
-            for arg in formatted_p.args:
-                dot.edge(arg.name, p)
-            dot.edge(p, formatted_p.ret.name)
+            if mode == 'both':
+                dot.node(p, formatted_p.name, shape='ellipse')
+                for arg in formatted_p.args:
+                    dot.edge(arg.name, p)
+                dot.edge(p, ret_name)
+            elif mode == 'task':
+                p = ret_name
+                dot.node(p, formatted_p.name, shape='ellipse')
+                for arg in formatted_p.args:
+                    dot.edge(arg.name, p)
+            elif mode == 'data':
+                for arg in formatted_p.args:
+                    if cluster and common_provider and '[' not in arg.name:
+                        # Avoid duplicate arrows to subnodes if all providers are the
+                        # same and the argument is not a generic
+                        if arg.name not in cluster_connected:
+                            dot.edge(
+                                arg.name,
+                                ret_name,
+                                lhead=subgraph.name,
+                                # Thick pen to indicate multiple connections
+                                penwidth='2.0',
+                            )
+                            cluster_connected.append(arg.name)
+                    else:
+                        dot.edge(arg.name, ret_name)
         # else: Do not draw dummy providers created by Pipeline when setting instances
 
 
```

Comment:
There is an error in the built docs (in parameter tables):
![Screenshot at 2024-10-30 09-24-00](https://github.com/user-attachments/assets/3fa75bf2-3696-400b-a395-90e36e5d0ad5)